BE-013 Liquibase (Elizaveta Dubogryz)

el-dub opened the pull request.

el-dub committed:
Add liquibase, spring-data-jpa and postgresql dependencies

el-dub committed:
Update .gitignore

el-dub committed:
Create application.yaml

el-dub committed:
Add tables creating

el-dub committed:
Add foreign key constraints

el-dub committed:
Create db.changelog-master.yaml v.0.1

el-dub committed:
Create db.changelog-master.yaml

el-dub committed:
Change country_id to city_id

SergeyKoval commented:
Please resolve conflicts with dev branch first
![image](https://user-images.githubusercontent.com/3354988/121491176-6509dd00-c9de-11eb-9b9c-0b81d9a500d5.png)

el-dub committed:
Merge branch 'dev' into liquibase

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-cities.yaml:
```diff
@@ -0,0 +1,19 @@
+      changes:
+        - createTable:
+            tableName: cities
+            columns:
```
City should have reference to country

el-dub replied:
+

el-dub committed:
Update pom.xml

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-coupons.yaml:
```diff
@@ -0,0 +1,29 @@
+            columns:
+              - column:
+                  name: id
+                  type: bigint
```
I suggest UUID as primary key here and in other tables

el-dub replied:
+

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-discounts.yaml:
```diff
@@ -0,0 +1,48 @@
+                  constraints:
+                    nullable: false
+              - column:
+                  name: end_time
```
I believe this field is optional

el-dub replied:
+

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-discounts.yaml:
```diff
@@ -0,0 +1,48 @@
+              - column:
+                  name: active
+                  type: boolean
+                  constraints:
```
better provide default value here

el-dub replied:
+

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-users.yaml:
```diff
@@ -0,0 +1,64 @@
+                  constraints:
+                    nullable: false
+              - column:
+                  name: country_id
```
Why do you need country here if you have city. City should have reference to country

el-dub replied:
+

el-dub committed:
Update db.changelog-discounts.yaml

Add default value for active
Delete nullable: false from start_time
Change id types to UUID

el-dub committed:
Update db.changelog-cities.yaml

Add country_id
Change id types to UUID

el-dub committed:
Delete references to country from vendor_locations and users

el-dub committed:
Change id types to UUID

Artem-Kishenets reviewed (dismissed):
Хорошо

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-coupons.yaml:
```diff
@@ -0,0 +1,28 @@
+                    nullable: false
+              - column:
+                  name: date
+                  type: date
```
why not tymestamp?

el-dub replied:
+

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-users.yaml:
```diff
@@ -0,0 +1,58 @@
+      id: 10-add-user-role-enum
+      author: elizaveta_dubogryz
+      dbms: postgresql
+      sql: CREATE TYPE USER_ROLE as ENUM ('USER', 'ADMIN');
```
I believe you have 3 roles, aren't you?

el-dub replied:
We finally decided with frontend team that we'll have 2 roles

Hanna-Kalashnyk replied:
Could we also rename Users into Persons and User-Role to Role?  
It will help to avoid a mess with security part, 
because the word USER is  assigned by Spring Boot Autontification module.

Hanna-Kalashnyk replied:
Also this way we'll allow to separate the role filtering ("findPeopleByRole endpointor" find "findAllUsers") from "findAllPeople search".
More over it's a good style - to name diferrent entityes with different names...

SergeyKoval replied:
Not critical, but could be renamed. You should decide it inside your team

SergeyKoval replied:
About 2 roles lets discuss on the status call

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-users.yaml:
```diff
@@ -0,0 +1,58 @@
+                  constraints:
+                    nullable: false
+              - column:
+                  name: telephone
```
```suggestion
                  name: phone
```

el-dub replied:
+

SergeyKoval commented on src/main/resources/db/changelog/v.0.1/db.changelog-vendor-locations.yaml:
```diff
@@ -0,0 +1,29 @@
+                  type: UUID
+                  constraints:
+                    nullable: false
+              - column:
```
I believe here you need country id as well and just one of them will be populated (either country or city). For example you have vendor, who works with all cities in Belarus

el-dub replied:
+

el-dub committed:
Add country_id in vendor_locations

el-dub committed:
Change date type to timestamp in coupons

el-dub committed:
Change telephone column to phone

Artem-Kishenets reviewed (approved):
The changes were agreed upon in advance

el-dub committed:
Update application.yaml

Artem-Kishenets reviewed (approved):
The changes were agreed upon in advance

Hanna-Kalashnyk commented on src/main/resources/db/changelog/v.0.1/db.changelog-users.yaml:
```diff
@@ -0,0 +1,58 @@
+                  constraints:
+                    nullable: false
+              - column:
+                  name: phone
```
Maybe, put here 
constraints:             
   nullable: true

Also about email. Some people are not sure to share their personality...

SergeyKoval replied:
all employees have corporate email

Hanna-Kalashnyk commented on src/main/resources/db/changelog/v.0.1/db.changelog-coupons.yaml:
```diff
@@ -0,0 +1,28 @@
+                    nullable: false
+              - column:
+                  name: date
+                  type: timestamp
```
This way, the certain time of coupon  retrieving would be generated automatically at POST query in service layer.
Maybe it could be     nullable: true?

SergeyKoval replied:
I don't think so

Hanna-Kalashnyk commented on .gitignore:
```diff
@@ -25,3 +25,7 @@ hs_err_pid*
+
+#Build
+
+target
```
Do U mind to include these, please:
# Operating System Files

*.DS_Store
.DS_Store
Thumbs.db


package.json
package-lock.json
webpack.config.js
webpack.generated.js
vscrudate.iml

# Eclipse Project Files #

.classpath
.project
.settings
.springBeans

# Test files

accessingdataneo4j.db
dependency-reduced-pom.xml
README.html
